enclave-runner/src/usercalls/mod.rs

```diff
@@ -165,18 +165,18 @@ pub trait SyncStream: 'static + Send + Sync {
 }
 
 
-trait SyncListener: 'static + Send + Sync {
-    fn accept(&self) -> IoResult<(FileDesc, Box<dyn ToString>, Box<dyn ToString>)>;
+pub trait SyncListener: 'static + Send + Sync {
+    fn accept(&self) -> IoResult<(Box<dyn SyncStream>, Box<dyn ToString>, Box<dyn ToString>)>;
```

Comment:
Should this return the socket addresses? Or should this be in line with what we do in `connect_stream/bind_stream`: take an argument `local_addr: Option<&mut String>, peer_addr: Option<&mut String>,`?

Reply:
I guess returning socket address might be ok, since in the rust std library we use `addr_to_sockaddr` for `peer_address` in other cases we do accept strings.

Reply:
I don't think you got my question. `String`s are correct. I was asking about the way it's passed.

Reply:
Sorry, mostly I sill did not get the question. I understand the part that passing string between runner and enclave is correct, but in case of accept passing a string that does not convert to a socket address leads to an error in the std library, some thing that does not happen say for connect_stream returning a non TCP address.

From that point of view, at the level of UserCallExtn/SyncListener, should we enforce that a SocketAddress is returned?

Reply:
My question is about the function signature: return value versus mutable reference argument.

Regarding your point: it doesn't matter what `std` does right now, we need to follow the [ABI](https://edp.fortanix.com/docs/api/fortanix_sgx_abi/struct.Usercalls.html#networking), which says that addresses are UTF-8 strings.

Reply:
Thanks @jethrogb .
I have made it consistent with connect_stream and pushed 2 new commits.
If the approach to accept an optional mutable string looks fine, I will fixup the commits as needed.